### PR TITLE
fix the job validation to allow using fractional GPU

### PR DESCRIPTION
**Pull request by william-wang**
Address issue #822

### CODE REVIEW

File: pkg/webhooks/admission/jobs/validate/admit_job.go [hunk 1]
```diff
@@ -257,6 +258,12 @@ func validateTaskTemplate(task v1alpha1.TaskSpec, job *v1alpha1.Job, index int)
 		if container.SecurityContext != nil && container.SecurityContext.Privileged != nil {
 			coreTemplateSpec.Spec.Containers[i].SecurityContext.Privileged = nil
 		}
+		if _, found := container.Resources.Limits["nvidia.com/gpu"]; found {
```
Comment (hzxuzhonghu) [inline pkg/webhooks/admission/jobs/validate/admit_job.go:261]: please add a comment, assume this is for gpu share

Comment (william-wang) [reply pkg/webhooks/admission/jobs/validate/admit_job.go:262]: the comment is added.

Comment (k82cn) [reply pkg/webhooks/admission/jobs/validate/admit_job.go:262]: is there any const for "nvidia.com/gpu"? if not, let's add a new one.

**Comment by k82cn** [comment]
> please add a ut

that's ok to have a seperate PR for ut :)

File: pkg/webhooks/admission/jobs/validate/admit_job.go (hunk 1 above)
Comment (hzxuzhonghu) [reply pkg/webhooks/admission/jobs/validate/admit_job.go:262]: not any const for it now. 

What about other GPU types, like ascend, AMD?

Comment (william-wang) [reply pkg/webhooks/admission/jobs/validate/admit_job.go]: Add the ascend-310. Others are not supportted so far.

**Comment by k82cn** [comment]
please also commit to master for next release :)

**Pull request merged**